pages/article/[articleLink].js

```diff
@@ -0,0 +1,160 @@
+import { useState, useEffect } from 'react';
+import { useRouter } from 'next/router';
+import Link from 'next/link';
+import Image from 'next/image';
+import classNames from 'classnames';
+
+import { getArticleDetail } from '../../firebase/api/articles';
+import { Layout } from '../../components/Layout';
+import { PortalLink } from '../../components/PortalLink';
+import { KeywordLink } from '../../components/KeywordLink';
+import { Loader } from '../../components/Loader';
+import Arrow from '../../public/img/arrow.svg';
+
+const Article = () => {
+  const router = useRouter();
+  const { articleLink } = router.query;
+
+  const [article, setArticle] = useState();
+
+  const [isDetailsOpen, setIsDetailsOpen] = useState(false);
```

Comment:
Probably no need for separation, as they are just simple declaration with no logic around